modules/snooker/src/main/ts/ephox/snooker/api/Generators.ts

```diff
@@ -166,12 +171,52 @@ const merging = function (generators: Generators) {
       // Remove any width calculations because they are no longer relevant.
       Css.remove(raw, 'width');
       Css.remove(cell, 'width');
+
+      const scope = Attribute.get(cell, 'scope');
+
+      if (Type.isString(scope)) {
+        Attribute.set(raw, 'scope', scope);
+      }
+
       return raw;
     };
   };
 
+  const merge = (cells: SugarElement[]) => {
+    const getScopeProperty = () => {
+      const baseScope = Attribute.get(cells[0], 'scope');
+      const scopes = [ 'row', 'col' ];
+
+      const isMixed = Arr.exists(cells.slice(1), (c) => {
+        const scopeOpt = Attribute.getOpt(c, 'scope');
+
+        return scopeOpt.fold(
+          Fun.never,
+          (scope) => Arr.contains(scopes, scope) && scope !== baseScope
+        );
+      });
```

Comment:
```suggestion
      const isMixed = Arr.exists(cells.slice(1), (c) => {
        const scopeOpt = Attribute.getOpt(c, 'scope');
        return scopeOpt.exists((scope) => scope !== baseScope && Arr.contains(scopes, scope));
      });
```
This just removes the unnecessary fold and does some minor optimisation.